src/scene/board-scene-controller.ts

```diff
@@ -1,114 +1,293 @@
+            this.doTurn(players[0].uuid);
         }
     }
 
-    private addPlayersToBoard(players: Player[]) {
-        this.players = players;
-        this.scene.addPlayers(players.map(p => p.uuid));
+    onSceneWake(_: any, minigameResult: ResultData): void {
+        super.onSceneWake(_, minigameResult);
+        if (!(this.aznopoly.isHost && this.minigameInprogress)) {
+            return;
+        }
+
+        this.minigameInprogress = false;
+        this.onMinigameResult(minigameResult);
+        this.doTurn(this.players[0].uuid);
     }
 
-    private startTurn(uuid: string) {
-        this.currentPlayerUuid = uuid;
+    private onMinigameResult(result: ResultData) {
+        this.aznopoly.connectedUuids.forEach(uuid => {
+            if (!result.playerWon.includes(uuid)) {
+                this.syncProxy.removeMoney(uuid, 100);
+                console.log("Player lost", uuid);
+            } else {
+                console.log("Player won", uuid);
+            }
+        });
+    }
 
-        if(uuid == this.aznopoly.uuid) {
-            this.scene.enableRollButton();
+    public removeMoney(uuid: string, amount: number) {
+        const player = this.getPlayer(uuid);
+        if (!player) {
+            console.error("Player not found");
+            return;
+        }
+        player.money -= amount;
+
+        this.scene.updatePlayerInfo(uuid, player);
+    }
+
+    public addTiles(uuid: string, tiles: number[]) {
+        const player = this.getPlayer(uuid);
+        if (!player) {
+            console.error("Player not found");
+            return;
+        }
+
+        tiles.forEach(value => {
+            player.tiles.push(value);
+        })
+
+        this.scene.updatePlayerInfo(uuid, player);
+    }
+
+    public buyProperty(uuid: string) {
+        const player = this.players.find(p => p.uuid == uuid);
+        if (!player) {
+            return;
         }
+
+        this.propertyHelper.buyProperty(uuid, player.position);
     }
 
     public onRollClick() {
         this.scene.disableRollButton();
-        this.syncProxy.doDiceRoll();
+        this.syncProxy.rollDice();
     }
 
-    private doDiceRoll() {
-        if (!this.aznopoly.isHost) {
+    public onTurnEnd(uuid: string) {
+        const currentIndex = this.players.findIndex(p => p.uuid == uuid);
+        if (currentIndex == this.players.length - 1) {
+            this.minigameInprogress = true;
+            this.syncProxy.startMinigame("minigame-roomba");
+        } else {
+            const nextPlayer = this.getNextPlayerIndex(uuid);
+            this.doTurn(this.players[nextPlayer].uuid);
+        }
+    }
+
+    public onPopUpClick(submit: boolean) {
+        if (!this.currentTurn) return;
+
+        if(this.aznopoly.uuid == this.currentTurn?.getPlayer()) {
+            this.scene.hideBuyTilePopUp();
+            if (submit) {
+                this.currentTurn.doBuyProperty()
+            }
+        }
+    }
+
+    public onPropertyBuy(uuid: string) {
```

Comment:
would call it executeBuyProperty and leave the checks wether it is possible to the caller.